### PR TITLE
set default serverPort when there's no applicationIndex

### LINKED ISSUE #19782: Upgrade to Spring Boot 3
##### **Overview of the feature request**

Start upgrading to Spring Boot 3.

##### **Motivation for or Use Case**

Selfishly, I need it for a demo at ApacheCon, but I'd also like to hack on it during the JavaOne Hackergarten. 

##### **Related issues or PR**

- https://github.com/jhipster/generator-jhipster/issues/19768
- https://github.com/jhipster/generator-jhipster/issues/19934
- https://github.com/jhipster/generator-jhipster/issues/19991
- https://github.com/jhipster/generator-jhipster/issues/20004
- https://github.com/jhipster/generator-jhipster/issues/20297

### PR DESCRIPTION
**Pull request by Tcharl**
fixes #19782

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/generator.mjs [hunk 1]
```diff
@@ -747,6 +749,7 @@ export default class JHipsterServerGenerator extends BaseApplicationGenerator {
     return this.asEndTaskGroup(this.delegateTasksToBlueprint(() => this.end));
   }
 
+  aaaaaaaa;
```
Comment (DanielFran) [inline generators/server/generator.mjs:752]:
```suggestion
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

**Comment by Tcharl** [comment]
Oups, sorry @DanielFran , it was a code point

File: generators/app/__snapshots__/generator.spec.mts.snap [hunk 2]
```diff
@@ -1144,7 +1144,7 @@ exports[`generator - app with microservice should match snapshot 1`] = `
   "searchEngineCouchbase": false,
   "searchEngineElasticsearch": false,
   "searchEngineNo": true,
-  "serverPort": 8081,
+  "serverPort": 8080,
```
Comment (mshima) [inline generators/app/__snapshots__/generator.spec.mts.snap:1147]: IFAIK microservice port should be 8081 by default

Comment (Tcharl) [reply generators/app/__snapshots__/generator.spec.mts.snap:1147]: why this small change in the code can lead to that side effect? Looked 3 times at the logic, it shouldn't impact this snapshot. Also I'm not sure that the port should be 8081 as there's no gateway described in the test.

Comment (Tcharl) [reply generators/app/__snapshots__/generator.spec.mts.snap]: Updated, thank you very much for the review!

**Comment by mshima** [comment]
It workaround the initial generation, but others configs and generators may still be broken, see https://github.com/jhipster/generator-jhipster/pull/20839

**Comment by Tcharl** [comment]
Both PRs are complementary and valid to my understanding.
The issue which is fixed by this one is an empty port entry in a spring configuration which makes configuration validation to fail: `http://localhost:`.

**Pull request closed without merging**